# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt
from scipy import stats

In [2]:
# "tobs" is "temperature observations"

#File to load 
file_to_load = "Resources/hawaii_measurements.csv"

#Read file and store into pandas DataFrame
hawaii_df = pd.read_csv(file_to_load)
hawaii_df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
hawaii_df.date = pd.to_datetime(hawaii_df.date, infer_datetime_format = True)
hawaii_df.head()

station       date  prcp  tobs
0  USC00519397 2010-01-01  0.08    65
1  USC00519397 2010-01-02  0.00    63
2  USC00519397 2010-01-03  0.00    74
3  USC00519397 2010-01-04  0.00    76
4  USC00519397 2010-01-06   NaN    73

In [4]:
# Set the date column as the DataFrame index
hawaii_df = hawaii_df.set_index(hawaii_df['date'])
hawaii_df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [5]:
# Drop the date column
hawaii_df = hawaii_df.drop(columns='date')
hawaii_df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [6]:
# Filter data for desired months
june_data = hawaii_df[hawaii_df.index.month == 6]
december_data = hawaii_df[hawaii_df.index.month == 12]

In [7]:
# Identify the average temperature for June
june_data.mean()

prcp     0.136360
tobs    74.944118
dtype: float64

In [8]:
# Identify the average temperature for December
december_data.mean()

prcp     0.216819
tobs    71.041529
dtype: float64

In [9]:
# Create collections of temperature data
june_temp = june_data.tobs
december_temp = december_data.tobs

In [10]:
# Run paired t-test
stats.ttest_ind(june_temp, december_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

Across all of Hawaii's stations, the mean temperatures in the months of June and December only differ by about 3.9 degrees Farhenheit. To support this further, a t-test was conducted where although the p-value was very low, the data is still statistically significant. Now, while the difference in temperature is statistically significant, the actual difference isn't, which indicates that travelling to Hawaii in June or December would both be enjoyable!